src/parser.rs

```diff
@@ -168,7 +168,7 @@ impl<'a> Parser<'a> {
     }
 
     fn next_pos(&self) -> usize {
-        self.cur.clone().next().map(|p| p.0).unwrap_or(self.input.len())
+        self.cur.clone().next().map_or(self.input.len(), |p| p.0)
```

Comment:
I personally prefer `map().unwrap_or()` as it's easier to read, could this stay as it was?